theano/compile/nanguardmode.py

```diff
@@ -187,6 +200,47 @@ def compile_gpu_func(nan_is_error, inf_is_error, big_is_error):
             cuda_compile_failed = True
 
 
+def f_gpua_min(inp):
+    dt = inp.dtype
+    ctx_name = _name_for_ctx(inp.context)
+    key = (dt, ctx_name)
+    f = f_gpua_min.cache.get(key, None)
+    if f is None:
+        guard_in = GpuArrayType(str(dt), (False,), context_name=ctx_name)()
+        f = theano.function([guard_in], T.min(guard_in),
+                            mode='FAST_RUN', profile=False)
```

Comment:
I think we should force the gpu, by adding the flag 'gpuarray' idem in the old back-end(optional).

Idem in the other function.